opm/simulators/wells/MultisegmentWellGeneric.cpp

```diff
@@ -63,8 +65,15 @@ MultisegmentWellGeneric(WellInterfaceGeneric& baseif)
     baseif.perfDepth().resize(baseif_.numPerfs(), 0.);
     for (size_t perf = 0; perf < completion_set.size(); ++perf) {
         const Connection& connection = completion_set.get(perf);
-        if (connection.state() == Connection::State::OPEN) {
+        if (connection.state() == Connection::State::OPEN && connection.attachedToSegment()) {
```

Comment:
`connection.attachedToSegment()` will already return `return (segment_number > 0);`. So the condition `if ( segment_index == -1)` will never be reached?

Reply:
Indeed, fixed. Classic copy & paste without thinking. :flushed: